The build will fail on non-Windows systems

When performing cross-platform compilation using `cross` on non-Windows systems, such as macOS or Linux errors related to Windows-specific crates may occur.

System Information:

```zsh
❯ sw_vers
ProductName:            macOS
ProductVersion:         26.3.1
ProductVersionExtra:    (a)
BuildVersion:           25D771280a
❯ uname -m
x86_64
```

Cargo and cross Version Information:

```zsh
❯ cross --version
cross 0.2.5 (19bc73f 2026-02-06)
[cross] warning: unable to get metadata for package
[cross] note: Falling back to `cargo` on the host.
cargo 1.94.1 (29ea6fb6a 2026-03-24)
```

Reproduction Steps: 

1. Create a new Cargo project using `cargo new test-winpty-rs`
2. Add the `winpty-rs` dependency
3. Compile by executing the command `cross build --target=x86_64-pc-windows-gnu`.

Cargo.toml:

```toml
[package]
name = "test-winpty-rs"
version = "0.1.0"
edition = "2024"

[dependencies]
winpty-rs = { version = "1" }
```

cross build output:

<details>
<summary>Expand</summary>

```zsh
❯ cross build --target=x86_64-pc-windows-gnu
   Compiling proc-macro2 v1.0.106
   Compiling unicode-ident v1.0.24
   Compiling quote v1.0.45
   Compiling windows-link v0.2.1
   Compiling autocfg v1.5.0
   Compiling serde_core v1.0.228
   Compiling zmij v1.0.21
   Compiling libc v0.2.185
   Compiling serde v1.0.228
   Compiling serde_json v1.0.149
   Compiling memchr v2.8.0
   Compiling windows-strings v0.5.1
   Compiling windows-result v0.4.1
   Compiling windows-threading v0.2.1
   Compiling itoa v1.0.18
   Compiling crossbeam-utils v0.8.21
   Compiling num-traits v0.2.19
   Compiling glob v0.3.3
   Compiling bitflags v2.11.1
   Compiling which v8.0.2
   Compiling crossbeam-channel v0.5.15
   Compiling syn v2.0.117
   Compiling windows-implement v0.60.2
   Compiling windows-interface v0.59.3
   Compiling serde_derive v1.0.228
   Compiling enum-primitive-derive v0.3.0
   Compiling windows-core v0.62.2
   Compiling windows-future v0.3.2
   Compiling windows-collections v0.3.2
   Compiling windows-numerics v0.3.1
   Compiling windows v0.62.2
error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
  --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:87:51
   |
87 |             } else if *iid == <windows_core::imp::IMarshal as windows_core::Interface>::IID {
   |                                                   ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
  --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:89:43
   |
89 |                 return windows_core::imp::marshaler(
   |                                           ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:255:51
    |
255 |             } else if *iid == <windows_core::imp::IMarshal as windows_core::Interface>::IID {
    |                                                   ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:257:43
    |
257 |                 return windows_core::imp::marshaler(
    |                                           ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:422:364
    |
422 | ...core::Interface >::IID { & mut ( * this ) . vtable as * mut _ as _ } else if * iid == < windows_core::imp::IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( ...
    |                                                                                                               ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:422:470
    |
422 | ...:IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( * this ) . vtable as * mut _ as * mut core::ffi::c_void ) , interface ) ; } else { core::ptr::null_mut ( )...
    |                                                                                                               ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:572:51
    |
572 |             } else if *iid == <windows_core::imp::IMarshal as windows_core::Interface>::IID {
    |                                                   ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:574:43
    |
574 |                 return windows_core::imp::marshaler(
    |                                           ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:752:364
    |
752 | ...core::Interface >::IID { & mut ( * this ) . vtable as * mut _ as _ } else if * iid == < windows_core::imp::IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( ...
    |                                                                                                               ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:752:470
    |
752 | ...:IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( * this ) . vtable as * mut _ as * mut core::ffi::c_void ) , interface ) ; } else { core::ptr::null_mut ( )...
    |                                                                                                               ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find type `IMarshal` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:927:377
    |
927 | ...core::Interface >::IID { & mut ( * this ) . vtable as * mut _ as _ } else if * iid == < windows_core::imp::IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( ...
    |                                                                                                               ^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `marshaler` in module `windows_core::imp`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/bindings.rs:927:483
    |
927 | ...:IMarshal as windows_core::Interface >::IID { ( * this ) . count . add_ref ( ) ; return windows_core::imp::marshaler ( core::mem::transmute ( & mut ( * this ) . vtable as * mut _ as * mut core::ffi::c_void ) , interface ) ; } else { core::ptr::null_mut ( )...
    |                                                                                                               ^^^^^^^^^ not found in `windows_core::imp`

error[E0425]: cannot find function `submit` in crate `windows_threading`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/async_spawn.rs:264:28
    |
264 |         windows_threading::submit(move || {
    |                            ^^^^^^ not found in `windows_threading`

error[E0425]: cannot find function `submit` in crate `windows_threading`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/async_spawn.rs:281:28
    |
281 |         windows_threading::submit(move || {
    |                            ^^^^^^ not found in `windows_threading`

error[E0425]: cannot find function `submit` in crate `windows_threading`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/async_spawn.rs:298:28
    |
298 |         windows_threading::submit(move || {
    |                            ^^^^^^ not found in `windows_threading`

error[E0425]: cannot find function `submit` in crate `windows_threading`
   --> /Users/secvision/.cargo/registry/src/mirrors.ustc.edu.cn-38d0e5eb5da2abae/windows-future-0.3.2/src/async_spawn.rs:315:28
    |
315 |         windows_threading::submit(move || {
    |                            ^^^^^^ not found in `windows_threading`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `windows-future` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...

What's next:
    Debug this container error with Gordon → docker ai "help me fix this container error"

```
</details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,14 +49,14 @@ features = [
     "Win32_System_WindowsProgramming"
 ]
 
-[build-dependencies.windows]
+[target.'cfg(windows)'.build-dependencies.windows]
 version = "0.62"
 features = [
     "Win32_Foundation",
     "Win32_System_LibraryLoader"
 ]
 
-[build-dependencies.windows-bindgen]
+[target.'cfg(windows)'.build-dependencies.windows-bindgen]
 version = "0.64"
 
 [dev-dependencies]

